integrations/kube-agent-updater/pkg/cache/cache_test.go

```diff
@@ -0,0 +1,91 @@
+package cache
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+)
+
+func failIfCalled(t *testing.T, value string, err error) func(context.Context) (string, error) {
+	return func(_ context.Context) (string, error) {
+		require.Fail(t, "should not be called")
+		return value, err
+	}
+}
+
+func failIfCalledTwice(t *testing.T, value string, err error) func(context.Context) (string, error) {
+	var fuse *bool
```

Comment:
Nit: Wouldn't this work without being a pointer? I'm fairly certain Go closures work fine w/o this being a pointer.